### DISCUSSION TITLE
/part-5-search-refiner-control-methods-explained/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#76] [Comments]
Part 5: The Search Refiner Control Methods Explained
        
[https://www.eliostruyf.com/part-5-search-refiner-control-methods-explained/](https://www.eliostruyf.com/part-5-search-refiner-control-methods-explained/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Andy on 2014-02-01 00:04
```

We were muddling around trying to figure this out ourselves until I searched and found this article. This is exactly what we were wanting to do and finding such a well written and documented explanation nearly made us cry. Thank you so much!

**Comment by comment-migrator** [comment]
```header
Originally posted by Anatoli on 2014-03-25 16:03
```

Hi Elio,

I've derived a custom control from ContentBySearchWebPart. Is there a possibility to extend my custom Content Search Web Part (CSWP) with a refiner based on a content type property?

To explain it in more detail: I'm trying to extend the standard CSWP with a custom refinement, for example as radio button list that is placed right above the search results.

While searching for information I've found your js documentation and thought if it would be possible to achieve this using the js methods?

Any idea?

Thanks!

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-03-26 19:22
```

Anatoli,

These functions only work for a refinement control. You could add a refinement web part to the page, and configure it to use your refinement property.

Regards,  
Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by Anatoli on 2014-03-26 19:26
```

Hi Elio,  
Thanks for your answer anyway.

**Comment by comment-migrator** [comment]
```header
Originally posted by Troels on 2014-05-21 03:37
```

Thanks for this beautiful and clear explanation. It always makes me happy when I find a no-crap lowdown on a SharePoint topic. For some reason, so many pages about SharePoint just says so little even if the talk is long enough to outgrow a sane patience. But not here, so thank you Elio!

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-05-22 00:56
```

Thanks for the feedback!

**Comment by comment-migrator** [comment]
```header
Originally posted by Mike on 2014-06-09 17:43
```

Hi Elio,  
An excellent series of posts - thank you for that.

I'm trying to create a refiner that will allow me to refine content based on the initial part of each hit's Path. I have a multi-site collection web app, but want to present four divisions of the available content. I started by building a refiner that works on the SPSiteURL property but I'll need to hard-code each of these mappings and with the number of values that match each division this is likely to be too many values for the URL. Ideally I'd want the JSON object in the URL to support a 'contains' search operator instead of an 'equal'. Do you know if this is possible, and if so how we'd go about doing this?

Thanks

**Comment by comment-migrator** [comment]
```header
Originally posted by alexeyww on 2014-07-10 19:50
```

Hi Elio,

Thank you for sharing your knowledge about display templates.  
Some things I need I've managed to find on your site only.

Could you please help me clarify one moment about Refinement filters:  
As refinement filters use url's query string to work it seems impossible to have several Refinement panels that utlilize refinement filters on the same page.

Is it correct?

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-07-15 15:46
```

alexeyww,

Thank you for the feedback.

You should be able to use multiple refinement web parts on the same page. They work perfectly together. Are you experiencing some problems with it?

Regards,  
Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by alexeyww on 2014-07-15 16:02
```

Elio,  
I did not make myself clear, sorry.

I ment that I needed to have several groups of search web parts (refinement, search results, search box) on the same page. And each group of search web parts should work independently from another group. I thought that if each group uses the same URL hash "Default" for coordination inside the group it would be problematic to achieve independence. Fortunately it appears that each group of search web parts can use their own URL hash parameter which can be set as web part property "QueryGroupName".

**Comment by comment-migrator** [comment]
```header
Originally posted by Mahesh Sasidharan on 2014-07-16 00:43
```

I found this blog a bit too late.. ;(  
Added all these functions using custom code... Devastated. But still a great blog.  
Does MSDN even have these things covered?

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-07-16 00:53
```

Mahesh Sasidharan,

Thanks.  
At the time I wrote this, no MSDN information was available about this topic and I think this is still the case.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Sandip on 2014-08-11 20:01
```

Hi Elio,

Thanks for your great post..It's really helpful.

I have the same problem as one asked by Mike i.e. I’d want the JSON object in the URL to support a ‘contains’ search operator instead of an ‘equal’. Do you know if this is possible, and if so how we’d go about doing this?

Thanks,  
Sandip

**Reply by comment-migrator** [reply]
```header
Originally posted by hyankov on 2014-12-15 21:47
```

this, please

**Reply by comment-migrator** [reply]
```header
Originally posted by hyankov on 2014-12-16 15:22
```

Sandip,

As far as I can tell, if you replace the hex token with a plain-text token, the search seems to work in a 'contains' mode.

[https://hyankov.wordpress.c...](https://hyankov.wordpress.com/2014/12/15/sharepoint-2013-refiner-multi-value-contains-instead-of-an-equals/ "https://hyankov.wordpress.com/2014/12/15/sharepoint-2013-refiner-multi-value-contains-instead-of-an-equals/")

**Comment by comment-migrator** [comment]
```header
Originally posted by Sharath on 2014-10-06 19:04
```

Hi,  
I have made use of Srch.Refinement.getExpanded and able to collapse and expand the refinement panel.  
Here the requirement is by default all the refiners should be in collapsed state. But selecting one refiner and click on Apply button, only that refiner category should be expandable.

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-10-17 20:48
```

Sharath,

To make them by default collapsed, you will need to update the display template and invert the **isExpanded** logic. By default it does this:

var displayStyle = (isExpanded == "true" ? "" : "none");

That should be inverted to:

var displayStyle = (isExpanded == "true" ? "none" : "");

For the implementation of the apply button, you will need to write some extra JavaScript to retrieve the exact refinement control, but you could use the **Srch.U.collapsibleRefinerTitle** show or hide the refinement control was you have done your check.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Christophe Vandeput on 2014-11-07 14:08
```

Elio,

How can I work with alias instead of RefinableString01 ?  
When I use the alias the filtering works but the values are not checked in the refinementpanel.

Kind Regards

Christophe

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-11-07 15:14
```

Christophe,

You would need to select the RefinableString01 as the refiner, instead of the alias. You could rename the refiner control title.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by hyankov on 2014-12-15 21:19
```

k: stands for ctx.RefinementControl\["csr\_useKQL"\]

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2014-12-16 00:50
```

hyankov,

Thank you for the information. Appreciated.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Vlad on 2015-03-31 22:41
```

Hi,

What if the managed property value contains two ore more separate words? In my case, I have a site column of type text and the crawled property for that column is mapped to a managed property.  
If i'm trying to construct my Json like this : var refinerValue = {'MyManagedProperty':\['string1 string2'\]}, my search results web part throws an error when calling the function 'addRefinementFilters' : "Property doesn't exist or is used in a manner inconsistent with schema settings."  
The refinement works fine like this: var refinerValue = {'MyManagedProperty':\['string1\*'\]}  
There are any syntax rules that I have to use?

Thx, Vlad

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-04-07 18:15
```

Hi Vlad,

You can pass an array as object to the refiner. Here is an example:

var refinerValue = {'Title':\['elio','struyf'\]}

That object can be passed to your refiner control:

$getClientControl(this).addRefinementFilters(refinerValue)

The refinement in the URL will look like this:

"r":\[{"n":"Title","t":\["elio","struyf"\],"o":"and","k":false,"m":null}\]

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Julian on 2015-05-06 01:35
```

FYI. I believe that the 'k' variable in the refiner (you say "still searching for the use of this") is the KQL flag.

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-05-06 05:16
```

Julian,

Still needed to update it. hyankov also mentioned it a couple of months ago.

Thanks for the info.

Regards,  
Elio

**Comment by comment-migrator** [comment]
```header
Originally posted by Garry Sinclair on 2015-05-08 02:09
```

In Internet Explorer, if too many options are selected in a multi-value refiner (based on a taxonomy field type), the size of the hash in the url gets too long (over 2kb), resulting in a javascipt error (i read this is the hash value limit in IE), and the refiner not working. Have you experienced this, and do you have any suggestions to get around this issue?

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-05-08 21:01
```

Garry,

The problem with the taxonomy fields is that you may not use the auto-created managed property. This auto-created managed property is mapped with the taxonomy ID, that is what is making your URL getting to long.

Microsoft has also mentioned it on a Technet article, but it is to well hidden. What you best do is create a new managed property or use the auto defined onces and map the ows\_column crawled property with it, instead of using the ows\_taxId\_column crawled property. I have already described the process in another blog post about sorting managed metadata properties: [Sorting search results by managed metadata properties](http://www.eliostruyf.com/correctly-sorting-search-results-by-managed-metadata-properties/ "Sorting search results by managed metadata properties"). In the **solution** section you find the process to map the right crawled property. Once you have this, you need to do a full crawl and use the new managed property as your refiner.

Regards,  
Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by Garry Sinclair on 2015-05-11 01:26
```

Hi Elio,

I'll give your suggestion a try. Unfortunately, we have quite a large number of taxonomy fields, and it will be a pain to map them all over, and then change the web parts to use the new managed properties as refiners, but if it works, it will be worth the effort!

Thanks for the quick reply :-)

Regards, Garry

**Comment by comment-migrator** [comment]
```header
Originally posted by Yogesh Ghare on 2015-05-21 11:37
```

Hi Elio,

I am looking to build a custom refiner from Managed Navigation. The refiner should be build the same way the taxonomy (tree view). I am refering to using Filter\_Default.html template to build the tree view refiner.

I have split the refiner values into arrays and then built the tree object by

var arrTemp = filter.RefinementName.split(":");  
addToTree(Tree,arrTemp);

var Tree = {};

function addToTree(tree, array) {  
for (var i = 0, length = array.length; i < length; i++) {  
tree = tree\[ array\[i\] \] = tree\[ array\[i\] \] || {};  
}  
}

But I am not sure on how to build the UL element and dynamically append the sub nodes to the parent LI (in the OutputFilter function)

Any reference would be highly appreciated.

Regards,  
Yogesh

**Reply by comment-migrator** [reply]
```header
Originally posted by Raj on 2016-10-07 22:33
```

Hi Yogesh, Did you get any solution for this ? I am also looking for the same requirement.

**Reply by comment-migrator** [reply]
```header
Originally posted by Preethi on 2020-03-03 04:02
```

Hi  
Did you found any solution.  
I am looking for the same.

Current Scenario:  
D1,D2,D3  
D2,D3,D4

I want to show the values as  
D1  
D2  
D3  
D4

**Reply by comment-migrator** [reply]
```header
Originally posted by Daniel on 2017-03-15 17:12
```

Hi,  
there is a webpart that provides a tree view refiner for your search page:  
\* for 2013/16 onpremise: [http://www.diqa-pm.de/en/Se...](http://www.diqa-pm.de/en/Semantic_Search_Webparts "http://www.diqa-pm.de/en/Semantic_Search_Webparts")  
\* for SP online: [https://store.office.com/en...](https://store.office.com/en-001/app.aspx?assetid=WA104380629&sourcecorrid=1887320b-13ca-4dc6-bb0f-a2b9a33e30ee&searchapppos=0&appredirect=false&omkt=en-001&ui=en-US&rs=en-001&ad=US "https://store.office.com/en-001/app.aspx?assetid=WA104380629&sourcecorrid=1887320b-13ca-4dc6-bb0f-a2b9a33e30ee&searchapppos=0&appredirect=false&omkt=en-001&ui=en-US&rs=en-001&ad=US")

**Comment by comment-migrator** [comment]
```header
Originally posted by naresh .k on 2016-06-06 07:18
```

Can we filter the date range using these methods. I tried $getClientControl(this).updateRefinersJSON('{"AstroArticleStartDate":\["range(2016-05-30T16:00:00.000Z,2016-05-19T16:00:00.000Z)"\]}') but nor working.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-06-06 12:23
```

You can indeed refine on date ranges, check out this post for more information about it: [http://www.eliostruyf.com/p...](http://www.eliostruyf.com/part-7-working-ranges-search-refiner-control/ "http://www.eliostruyf.com/part-7-working-ranges-search-refiner-control/")

If it doesn't work, check if the MP is refinable and type set to datetime. You can also check for errors in the browser developer console.

**Reply by comment-migrator** [reply]
```header
Originally posted by naresh .k on 2016-06-07 16:54
```

Its working..Great work.. Helped me lot in achieving most of my tasks .. Wonderfull blog.. Keep posting..

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-06-07 19:33
```

Thanks for the feedback!

**Comment by comment-migrator** [comment]
```header
Originally posted by Dan A on 2016-10-13 19:51
```

Hi Elio, great article as always! I had a quick question that I was hoping you could shed some light on. Is there a special syntax or encoding that I need to perform for terms which contains spaces or multiple words?

For example I am passing "Speciality Companies" but the space between terms blows up my refiner. I tried encoding the strings but that didn't seem to work either :/. If I remove that term it works perfectly, I just can't seem to figure out how to use terms which are a phrase or multiple words. Any Ideas?

var refiner = $getClientControl(document.getElementById("Refinement"))  
var refinerValue = {'owstaxIdCNCMainAudience':\['Fidelis', 'Envolve', 'Speciality Companies'\]}  
refiner.addRefinementFiltersWithOp(refinerValue, 'or')

Thanks Again!!!  
Dan

**Reply by comment-migrator** [reply]
```header
Originally posted by Dan A on 2016-10-13 20:40
```

Well after some trial and error I was able to get this working. I was over thinking it, I simply had to wrap the phrase in double quotes. Hope this helps someone else :).

var refinerValue = {'owstaxIdCNCMainAudience':\['Fidelis', 'Envolve', '"Speciality Companies"'\]}

Thanks Again,  
Dan

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-10-17 11:03
```

Thanks for sharing your solution.

**Comment by comment-migrator** [comment]
```header
Originally posted by Basant Pandey on 2016-12-19 09:06
```

Hi Elio,

Thanks for this great post. I have one question related to the refiners maximum values option. Here is the scenario. I have created the custom refiner typeahead in a refiner, I mapped one custom column with one Managed property and make it to refinable as attribute. The list having 5000 items all are unique and the refiner only showing 1994 items. Can you please why this showing this much count I am expecting more than this. It would be great help me if you have any idea about the search boundaries related to this.

Thanks Basant

**Reply by comment-migrator** [reply]
```header
Originally posted on 2016-12-19 21:23
```

How are you doing your call for retrieving the refiners? Is this a typical search query? If so, check that you are retrieving duplicates. Most of the time, this is causing the trouble.

**Comment by comment-migrator** [comment]
```header
Originally posted by Iqbal sheikh on 2017-03-30 15:47
```

![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/703a5958c729b7b95db357206c46537269bbef68b077eb98f36f02b9f9aaa8ab.png)  
Hi Elio,  
I am new in SharePoint. I have created one page with refinement and search result. And on page load changing QueryText like below.  
var newQuery = '(ContentType:"Item" Path:"[https://xyz.com/sites/41190/](https://xyz.com/sites/41190/)"){SearchBoxQuery}';  
var appManager = Srch.ScriptApplicationManager.get\_current();  
var query = appManager.queryGroups\['Default'\];  
query.dataProvider.set\_queryTemplate(newQuery);  
query.dataProvider.issueQuery();  
It is updating searchresults data. But not updating left side (Entities) refiners web parts value.  
How can I update refiners value base on QueryText.

Please provide any suggestion.

Thankx

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-04-04 13:40
```

Is there any reason why you are updating the query on page load? You can also specify this query in the query builder so that you do not trigger a new query.

**Reply by comment-migrator** [reply]
```header
Originally posted by Iqbal sheikh on 2017-05-09 11:20
```

HI Elio,  
In our requirement we are getting path from other resource( path could be different) according to those path need to set refiner and search result dynamically.

Thanks  
Iqbal

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-05-22 13:49
```

Would query string params be an option? This can easily be included in the query.

**Comment by comment-migrator** [comment]
```header
Originally posted by Kevin Murphy-Ferguson on 2017-05-25 14:03
```

Hi Elio,  
What a fantastic resource this series of articles is! Can you tell me if a refiner can influence the sort order of the results? I have been changing the sort order of my results by using the "o" parameter that is a sibling of the "k" and the "r" parameters ([https://goo.gl/pFuFGQ)](https://goo.gl/pFuFGQ) "https://goo.gl/pFuFGQ)"), but doing that as a custom control that wrote out the whole Url. I would like to stop having to code the whole URL, and instead put my custom sorting logic into a refiner.  
Can a refiner have access or change the "o" parameter in the Url?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-05-30 15:29
```

There is always a way to do it, but by default the refiner methods can only be used for refining/filtering results. You cannot use it for sorting the results. If you want to make it happen, you are going for a similar approach you are already using.

**Comment by comment-migrator** [comment]
```header
Originally posted by jAJ on 2017-10-25 16:59
```

Hi Elio,  
Thanks for this great post !. When ever i perform a refiner using the function I dont see my Url getting updated with the #Default=... Do you know why ?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2017-10-31 11:52
```

Do you see any errors in the console? Do you see any changes happening in the search web part?

**Comment by comment-migrator** [comment]
```header
Originally posted by Dan on 2017-12-04 14:12
```

Hello,

this webpart can be embedded into the search page and displays the tree of terms:

[http://www.diqa-pm.de/en/Se...](http://www.diqa-pm.de/en/Semantic_Search_Webparts "http://www.diqa-pm.de/en/Semantic_Search_Webparts")

Regards

**Comment by comment-migrator** [comment]
```header
Originally posted by Sudheer on 2018-02-27 16:04
```

Hi Elio,  
I am trying to preset a refiner on SharePoint 2013 search.I have used multivalue display template and added the multiselect. Now I have a requirement to select a single value from one of the dropdown to be enabled only on page load, but remove it when users want to select some other refiner like as usually works. I was able to use 'get labels' method and refresh the multiselect. But unless I click on the dropdown the filter is not applying to search results.Could you please help, where to keep the jQuery("#" + "FileType" + "Select").multiselect("getLabels").find("input\[value='ǂǂ786c7378'\]").click(); for this to apply the filter to the search results on only first time page load.  
For the code please see here. Couldn't post here as it was a lot.  
[https://sudheersharepointst...](https://sudheersharepointstarters.blogspot.com/2018/02/multivalue-dropdown-refiner-sharepoint.html "https://sudheersharepointstarters.blogspot.com/2018/02/multivalue-dropdown-refiner-sharepoint.html")

**Comment by comment-migrator** [comment]
```header
Originally posted by Sudheer on 2018-03-01 19:22
```

Hi Elio,  
Thanks for the post. I am trying to append the #Default = query to the url.  
The requirement is to set the search value to what users have entered into the search box when landing to the search center instead of k:\*  
How to make k:{search box term} ?  
#Default={"k":"{user query term}","r":\[{"n":"Filetype","t":\["\\"%C7%82%C7%82786c7378\\""\],"o":"or","k":false,"m":null}\]}

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-03-02 11:56
```

The default searchbox should do this by default, are you creating one yourself? If that is the case, you can also create a display template for creating a customized seachbox experience.

**Reply by comment-migrator** [reply]
```header
Originally posted by Sudheer on 2018-03-02 14:24
```

My requirement is to have a refiner preselected when the page first loads, so that users can see the results only from that page.Now I can acheive this with either addrefinementfilter method or by appending the above k:\* to url.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-03-02 15:22
```

You can indeed work with the add refinement filter method.

Another way is to create your own query state, here is an example: [https://github.com/SPCSR/Di...](https://github.com/SPCSR/DisplayTemplates/blob/b1b0696992651af6ccc232055a9f56f3d3595e9a/Search%20Display%20Templates/Result%20refinement%20display%20templates%20(CSWP)/Control_Refine.html#L27-L46 "https://github.com/SPCSR/DisplayTemplates/blob/b1b0696992651af6ccc232055a9f56f3d3595e9a/Search%20Display%20Templates/Result%20refinement%20display%20templates%20(CSWP)/Control_Refine.html#L27-L46") - check the highlighted lines of code.

**Comment by comment-migrator** [comment]
```header
Originally posted by Kate on 2018-07-17 14:16
```

Hi Elio,

I need to create a "Clear all refiners" button.

I put the following code in JavaScript:

$getClientControl(document.getElementById('ctl00\_ctl40\_g\_eaaedfa7\_41a3\_4371\_ac28\_b6e8931cdbc7\_csr1')).updateRefinersJSON('{\\u0022RefinableString00\\u0022:null}');  
$getClientControl(document.getElementById('ctl00\_ctl40\_g\_eaaedfa7\_41a3\_4371\_ac28\_b6e8931cdbc7\_csr2')).updateRefinersJSON('{\\u0022RefinableString02\\u0022:null}');  
$getClientControl(document.getElementById('ctl00\_ctl40\_g\_eaaedfa7\_41a3\_4371\_ac28\_b6e8931cdbc7\_csr3')).updateRefinersJSON('{\\u0022RefinableString01\\u0022:null}');

To remove all 3 refiners and reset all.

It is working. However, after sometime (I think after each SharePoint full crawl, the refiners' id will change!  
So, my code does not work.

The question is, can I use getElementByName instead of getElementById ?  
e.g. $getClientControl(document.getElementByName("Group")).updateRefinersJSON('{\\u0022RefinableString00\\u0022:null}');

Is there any other way to clear all refiners?

Many thanks, Kate

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-07-17 22:26
```

Hi Kate, check out the following template which has a clear all functionality in place: [https://github.com/SPCSR/Di...](https://github.com/SPCSR/DisplayTemplates/tree/master/Search%20Display%20Templates/Refiners/Clear%20all%20refiners%20control%20template "https://github.com/SPCSR/DisplayTemplates/tree/master/Search%20Display%20Templates/Refiners/Clear%20all%20refiners%20control%20template")

**Reply by comment-migrator** [reply]
```header
Originally posted by Kate on 2018-07-18 09:35
```

Brilliant  
Thank you Elio, it is working.

**Comment by comment-migrator** [comment]
```header
Originally posted by Nathan Dutton on 2020-02-24 19:52
```

Hi Elio  
Im looking to have the refiners perform an OR logical condition, instead of an AND, by clicking on several refiner values like a multi value refinement item without having to click the apply button, is this doable?

**Comment by comment-migrator** [comment]
```header
Originally posted by Melih Eroglu on 2020-11-06 16:08
```

Hi Elio,  
Not 100% related to this topic but do you have a trick to read multivalue fields from query string and use them in KQL. EX:  
{|MultiValueCatagoryTaxManagedProperty:{QueryString.CategoryMultiValue}}

**Comment by tmGithub2021** [comment]
This is very helpful!  I'm having an issue when adding the line starting with "k".  It works without that, but I get an Unexpected Syntax error. I intend to use it with a custom refiner - to show the value as selected on the refiner.

<script>
$(document).ready(function() {
   var refiner = $getClientControl(document.getElementById("ctl00_ctl46_g_4fe5fff7_2e3f_43e4_bb83_a09eb1276daf_csr"));
   var refinerValue = {'FileType':['PDF']};
   refiner.addRefinementFilters(refinerValue);
   {"k":"test","r":[{"n":"FileType","t":["PDF"],"o":"and","k":false,"m":null}]}; - WORKS UNTIL I ADD THIS LINE
});
</script>

**Reply by tmGithub2021** [reply]
Responding to my own question about using the line starting with "k".  It seems that line might not be necessary if you convert the value to filter by to the hex value.  Use the below function to convert the value to hex:

function String2Hex(tmp) {
    var str = '';
    for(var i = 0; i < tmp.length; i++) {
        str += tmp[i].charCodeAt(0).toString(16);
    }
    return str;
}
